components/CourseFilter.tsx

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react';
+
+type CourseFilterProps = {
+  handleCategoryChange: (category: string) => void;
+};
+
+const CourseFilter: React.FC<CourseFilterProps> = ({ handleCategoryChange }) => {
+  const [selectedCategory, setSelectedCategory] = useState('All');
+
+  const handleSelectChange = (category: string) => {
+    setSelectedCategory(category);
+    handleCategoryChange(category);
+  };
+
+  return (
+    <div className="w-1/2 mb-20">
+      <select
+        className="w-full border-2 border-slate-400 p-2 rounded-full focus-within:border-cyan-600 transition-colors"
```

Comment:
This can be collapsed without the need of a wrapping div, and instead you can use a React fragment here to wrap it instead.

```suggestion
    <>
      <select
        className="w-1/2 mb-20 border-2 border-slate-400 p-2 rounded-full focus-within:border-cyan-600 transition-colors"
```